### DISCUSSION TITLE
flutter sample does not work

### DISCUSSION THREAD

**Discussion by jiangok-open** [supabase/supabase#15814] [Questions]
hi,

Sorry for this newbie question. I followed https://supabase.com/docs/guides/getting-started/quickstarts/flutter to try flutter+supabase but the flutter client cannot get data from supabase (the UI always shows a circling icon). The same url and anonKey works fine for the next.js sample. Any idea is highly appreciated.

Below is the sample code:

```
import 'package:flutter/material.dart';
import 'package:supabase_flutter/supabase_flutter.dart';

Future<void> main() async {
    WidgetsFlutterBinding.ensureInitialized();

  await Supabase.initialize(
      url: 'my_url',
      anonKey:
          'my_anon_key',
      );
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Countries',
      home: HomePage(),
    );
  }
}

class HomePage extends StatefulWidget {
  const HomePage({super.key});

  @override
  State<HomePage> createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  final _future = Supabase.instance.client
      .from('countries')
      .select<List<Map<String, dynamic>>>();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: FutureBuilder<List<Map<String, dynamic>>>(
        future: _future,
        builder: (context, snapshot) {
          if (!snapshot.hasData) {
            return const Center(child: CircularProgressIndicator());
          }
          final countries = snapshot.data!;
          return ListView.builder(
            itemCount: countries.length,
            itemBuilder: ((context, index) {
              final country = countries[index];
              return ListTile(
                title: Text(country['name']),
              );
            }),
          );
        },
      ),
    );
  }
}
```

supabase_flutter: ^1.6.2

**Comment by AkmenZ** [comment]
You shouldn't have <List<Map<String, dynamic>>> in the select statement. Just select();

**Comment by jiangok-open** [comment]
Thanks @AkmenZ. Sounds the sample code is wrong.

After using select(), the sample throws error:

lib/main.dart:44:17: Error: The argument type 'PostgrestFilterBuilder<dynamic>' can't be assigned to the parameter type 'Future<List<Map<String, dynamic>>>?'.

Any idea? Thanks.

**Reply by jiangok2006** [reply]
It turns out I run the code using macos app which need some [extra config](https://stackoverflow.com/questions/65458903/socketexception-connection-failed-os-error-operation-not-permitted-errno-1). Running using chrome works fine. Thanks.

**Reply by Tomyshh** [reply]
I'm on Windows and got the same issue : 

Error: The argument type 'PostgrestFilterBuilder' can't be assigned to the parameter type 'Future<List<Map<String, dynamic>>>?'.
